### PR TITLE
Radiopropa/minimizer rt

### PR DESCRIPTION
**Pull request by boboeyen**
This Pull Request will introduce a minimizer for the radiopropa propagation module and make it accessible to use it. This minimizing mode however is not working accurately at the moment and people are encouraged to resolve this problem.

Some addition are made in the config file to adjust some setting for the radiopropa module

### CODE REVIEW

**Review by shallmann — DISMISSED** [review]

**Comment by shallmann** [comment]
@boboeyen what do you mean by: "This minimizing mode however is not working accurately at the moment and people are encouraged to resolve this problem." ... is the minimizer not working reliably at all (if so why? and how can this be improved?), or are you referring to the missing functionality to minimize for bottom reflections? (in the latter case the logger info should be sufficient)

File: NuRadioMC/SignalProp/radioproparaytracing.py [hunk 1]
```diff
@@ -359,9 +369,131 @@ def delta(ray_dir,shower_dir):
...
+
+        def shoot_ray(theta):
+            ray = get_ray(theta,phi_direct)
+            sim.run(ray, True)
+            return ray
+
+        def cot(x):
+            return 1/np.tan(x)
+
+        def tan(x):
```
[the remark this answers is not in the document]
Comment (boboeyen) [reply NuRadioMC/SignalProp/radioproparaytracing.py:439]: ah yeah, the def of tan can be removed indeed

**Comment by boboeyen** [comment]
> @boboeyen what do you mean by: "This minimizing mode however is not working accurately at the moment and people are encouraged to resolve this problem." ... is the minimizer not working reliably at all (if so why? and how can this be improved?), or are you referring to the missing functionality to minimize for bottom reflections? (in the latter case the logger info should be sufficient)

The minimizer works in that sence that you can run it, but in a lot of cases the accuracy is bad and it is not that much faster than my iterative method. However, Christian proposed to make it accessible for others to use it in the interface but not recommand it. I have tried to make it more accurate during my week at Uppsala with Christian but couldn't figure out how to. Maybe others have some ideas?

**Review by anelles — APPROVED** [review]
I think it is good to move this into the main code. I think the caveats written are good enough. Since it is not used as default, it is fine for me to merge.

**Pull request merged**